filters.py

```diff
@@ -47,13 +56,16 @@ def process(new_data):
 
 
 def simple_predictive_filter(prediction_factor, velocity_smoothing_factor):
-	prev_prediction = Point(0, 0)
+	prev_prediction = None
 	est_velocity = Point(0, 0)
 	
 	def process(new_data):
 		nonlocal prev_prediction
 		nonlocal est_velocity
 		
+		if prev_prediction is None:
+			prev_prediction = new_data
+
```

Comment:
I know "picking up" isn't a thing at the moment, but do you think setting `est_velocity` to `Point(0,0)` in here as well would be a good idea?
I'm not sure, so if you think it's a good idea, do it. If not, it's fine.

Reply:
I'd considered it, but felt that leaving `est_velocity` initialized at the function beginning was a bit more clear. I'm not against moving it though if you feel like it later :)